### ISSUE TITLE
Clarify MET Compile Time Python requirements

### ISSUE THREAD

**Issue by DanielAdriaansen** [dtcenter/MET#2490]
This is a sub-issue of #2414.

The [current Appendix F documentation says](https://met.readthedocs.io/en/latest/Users_Guide/appendixF.html#compiling-python-support):

> The local Python installation must also support a minimum set of required packages. The MET build includes some python wrapper scripts to facilitate the passing of data in memory as well as the reading and writing of temporary files. The packages required by those wrapper scripts are **sys**, **os**, **argparse**, **importlib**, **numpy** and **netCDF4**. While most of these are standard packages and readily available, numpy and netCDF4 may not be. Users are advised to confirm their availability prior to compiling MET with python embedding support.

Questions for discussion:
1. Should we add a requirement for Xarray?
- If we add a requirement for Xarray, then [what of Xarray's optional dependencies](https://docs.xarray.dev/en/stable/getting-started-guide/installing.html#optional-dependencies) should we also include? Note that the `xarray.open_mfdataset` function requires `dask` to be installed.
- What is the current **minimum version of Python** that should be used? 3.8.6? 3.10.X?

Action items:

- [x] Remove standard Python modules **os**, **sys**, and **importlib**

**Comment by hsoh-u** [comment]
The API which uses "importlib" module/package is added to scripts/python/pyembed/python_embedding.py (#2285 ). "import importlib.util" can be removed by importing the pyembed module/package and by calling pyembed_tools.call_python(argv)

Here is an example:
```
try:
    from python_embedding import pyembed_tools
except:
    from pyembed.python_embedding import pyembed_tools

#pyembed_tools.add_python_path(__file__)    # un-comment this if there is a path problem
if __name__ == '__main__':
    met_in = pyembed_tools.call_python(sys.argv)
```

**Comment by DanielAdriaansen** [comment]
> The API which uses "importlib" module/package is added to scripts/python/pyembed/python_embedding.py (#2285 ). "import importlib.util" can be removed by importing the pyembed module/package and by calling pyembed_tools.call_python(argv)

Thanks Howard. I think since "importlib" is part of the Python standard lib, we can still use it if we want. Also, I've removed all Python packages that are part of the standard lib, since if the user has Python installed it should have those packages by default.

**Comment by DanielAdriaansen** [comment]
Documentation updates included in #2509 to address this.
